lib/elixir/lib/option_parser.ex

```diff
@@ -492,4 +556,22 @@ defmodule OptionParser do
       :unknown
     end
   end
+
+  defp format_errors(errors) do
+    details = Enum.map_join(errors, ". ", &format_error(&1))
+    total = Enum.count(errors)
+    error_pluralize = pluralize(total, "error", "errors")
+    "You have #{total} #{error_pluralize}! #{details}"
+  end
+
+  defp format_error({option, nil}) do
+    "The option #{option} does not exist"
+  end
+
+  defp format_error({option, value}) do
+    "The option #{option} exists but it is of the wrong type: #{inspect value}"
```

Comment:
What if we bring the information about `strict` down here to `format_error/1`, so we can have an error message like:

```elixir
"Option #{option} is of the wrong type; expected a #{type}, given #{inspect value}"
```